lms/services/course.py

```diff
@@ -73,6 +75,39 @@ def upsert_course(self, context_id, name, extra, settings=None) -> Course:
             type_=Grouping.Type.COURSE,
         )[0]
 
+    def find_group_set(self, group_set_id=None, name=None, context_id=None):
+        """
+        Find group sets stored in Courses of this applications instance.
+
+        Group sets are stored as part of Course.extra, this method allows to query and filter them.
+
+        :param context_id: Match only group sets of courses with this ID
+        :param name: Filter courses by name
+        :param group_set_id: Filter courses by ID
+        """
+        group_set = (
+            func.jsonb_to_recordset(Course.extra["group_sets"])
+            .table_valued(
+                column("id", Text), column("name", Text), joins_implicitly=True
```

Comment:
The names here make everything look more complicated "joins_implicitly", "render_derived"

We are just calling `jsonb_to_recordset` which is pretty a nice.